front/app/hooks/useExceedsSeats.ts

```diff
@@ -0,0 +1,52 @@
+import useSeats from 'api/seats/useSeats';
+import { isNil } from 'utils/helperUtils';
+import { useAvailableSeats } from './useAvailableSeats';
+
+interface ReturnFunctionParameters {
+  newlyAddedAdminsNumber: number;
+  newlyAddedModeratorsNumber: number;
+}
+
+interface ExceedsSeats {
+  admin: boolean;
+  moderator: boolean;
+  any: boolean;
+  all: boolean;
+}
+
+export const useExceedsSeats = (): ((
+  params: ReturnFunctionParameters
+) => ExceedsSeats) => {
+  const availableSeats = useAvailableSeats();
+  const { data: seats } = useSeats();
+
+  if (isNil(seats) || isNil(availableSeats))
+    return (_params: ReturnFunctionParameters) => ({
+      admin: false,
+      moderator: false,
+      any: false,
+      all: false,
+    });
+
+  const { availableAdminSeats, availableModeratorSeats } = availableSeats;
+
+  return ({
+    newlyAddedAdminsNumber,
+    newlyAddedModeratorsNumber,
+  }: ReturnFunctionParameters) => {
+    const exceedsAdminSeats = availableAdminSeats
+      ? seats.data.attributes.admins_number + newlyAddedAdminsNumber >
```

Comment:
If we move this line and declare a named const for it above, it could make this code easier. We could do something similar for line 42 maybe

`const expectedNewAdminSeats = seats.data.attributes.admins_number + newlyAddedAdminsNumber`

Reply:
Good point, done.